src/script/interpreter.cpp

```diff
@@ -503,6 +504,7 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                     if (vfExec.empty())
                         return set_error(serror, SCRIPT_ERR_UNBALANCED_CONDITIONAL);
                     vfExec.pop_back();
+                    fExec = fExec ? true : !count(vfExec.begin(), vfExec.end(), false);
```

Comment:
in `OP_ENDIF`, if `fExec` is true, the original `vfExec` must be by definition all true. Since `OP_ENDIF` will simply remove the last `vfExec` item, the resulting `vfExec` must not have any false. So we know `fExec` must remain true.

If `fExec` is false, there is at least one false in `vfExec` but we don't know where is it, so we have to count as usual